phi/model/openai/chat.py

```diff
@@ -323,261 +378,167 @@ def response(self, messages: List[Message]) -> ModelResponse:
                 function_calls_to_run.append(_function_call)
 
             if self.show_tool_calls:
-                if len(function_calls_to_run) == 1:
-                    model_response.content += f"\n - Running: {function_calls_to_run[0].get_call_str()}\n\n"
-                elif len(function_calls_to_run) > 1:
-                    model_response.content += "\nRunning:"
-                    for _f in function_calls_to_run:
-                        model_response.content += f"\n - {_f.get_call_str()}"
-                    model_response.content += "\n\n"
+                calls_str = "\n".join(f" - Running: {_f.get_call_str()}" for _f in function_calls_to_run)
+                model_response.content += f"\n{calls_str}\n\n"
 
             function_call_results = self.run_function_calls(function_calls_to_run)
             if len(function_call_results) > 0:
                 messages.extend(function_call_results)
-            # -*- Get new response using result of tool call
+            return model_response
+        return None
+
+    def _create_agent_message(
```

Comment:
same, can be without the leading `_`

Also should the be `create_assistant_message` as the role here is `assistant` ?